src/include/duckdb/catalog/catalog_entry/schema_catalog_entry.hpp

```diff
@@ -62,6 +63,8 @@ class SchemaCatalogEntry : public CatalogEntry {
 	CatalogSet sequences;
 	//! The catalog set holding the collations
 	CatalogSet collations;
+	//! The catalog set holding the enums
+	CatalogSet enums;
```

Comment:
I think we should all talk about whether this is the right model, or whether it should be a mapping of named types?

Reply:
Just to be sure we are on the same page.

Instead of having a  CatalogSet of specifically ENUMs, we have one for all the user_types, and these can hold CatalogEntries for the different user types (Enums and what-not right?) Or should the Catalog Entries also be sufficiently generic for any user type?

On another note, I've briefly chatted with Hannes and Mark yday in the office, I think in the next PR I will go on having the enum types living only in the catalog, adding the dependencies to the tables unless they are created directly from one of the other interfaces that not SQL (e.g., python,r, parquet), so we can be a bit more consistent at least on the types created though the SQL layer :)

Reply:
> Instead of having a CatalogSet of specifically ENUMs, we have one for all the user_types, and these can hold CatalogEntries for the different user types (Enums and what-not right?) Or should the Catalog Entries also be sufficiently generic for any user type?

I don't think I have an opinion on this part, but I think since enums are types and we would have types in the catalog, it just seems simpler to store type descriptions.

Reply:
I agree, I think it would indeed be simpler for future extensions if the `EnumCatalogEntry` is changed to a `TypeCatalogEntry` that holds only a `LogicalType`. That way adding additional functionality for `CREATE TYPE` should be rather trivial, as it could reuse all of the enum infrastructure. The only challenge I see here would be with the `ToSQL`, but this could have a switch based on the type (e.g. for `LogicalTypeId::ENUM`, output `CREATE ENUM`, otherwise for struct output `CREATE TYPE`, etc).